src/main/kotlin/com/jetbrains/snakecharm/lang/documentation/SmkShadowSettingsDocumentation.kt

```diff
@@ -0,0 +1,42 @@
+package com.jetbrains.snakecharm.lang.documentation
+
+import com.intellij.lang.documentation.AbstractDocumentationProvider
+import com.intellij.openapi.editor.Editor
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import com.intellij.psi.util.PsiTreeUtil
+import com.jetbrains.python.psi.PyStringLiteralExpression
+import com.jetbrains.snakecharm.lang.psi.SMKRuleParameterListStatement
+
+
+class SmkShadowSettingsDocumentation : AbstractDocumentationProvider() {
+    override fun generateDoc(
+            element: PsiElement?,
+            originalElement: PsiElement?): String? {
+        if (!element.isInShadowSection() || element?.parent !is PyStringLiteralExpression) {
```

Comment:
Before going through parent nodes, please ensure that you are in snake make file, not in *.py file. E.g. using
```
SnakemakeLanguageDialect.isInsideSmkFile(element)
```